### PR TITLE
ref(pageFilters): Return stable reference from getState

### PR DESCRIPTION
**Pull request by scttcper**
- removes unused hasInitialState
- mashes everything into a readonly state object that is updated on state change

part of https://github.com/getsentry/frontend-tsc/issues/66

### CODE REVIEW

File: static/app/stores/pageFiltersStore.spec.tsx [hunk 1]
```diff
@@ -36,6 +36,12 @@ describe('PageFiltersStore', function () {
     });
   });
 
+  it('returns a stable reference with getState', () => {
+    PageFiltersStore.updateProjects([1], []);
+    const state = PageFiltersStore.getState();
+    expect(Object.is(state, PageFiltersStore.getState())).toBe(true);
```
Comment (ryan953) [inline static/app/stores/pageFiltersStore.spec.tsx:42]: This assertion could be `.toBe()` which uses `Object.is` internally, and will give better messages if the assertion fails

https://jestjs.io/docs/expect#tobevalue

File: static/app/stores/pageFiltersStore.tsx [hunk 2]
```diff
@@ -117,45 +101,46 @@ interface PageFiltersStoreDefinition
 }
 
 const storeConfig: PageFiltersStoreDefinition = {
-  selection: getDefaultSelection(),
-  pinnedFilters: new Set(),
-  desyncedFilters: new Set(),
-  shouldPersist: true,
-  hasInitialState: false,
+  state: {
```
Comment (ryan953) [inline static/app/stores/pageFiltersStore.tsx:104]: could be using the `private` or `protected` keyword so callers know to continue to use `getState()`

```suggestion
  private state: {
```

Comment (ryan953) [reply static/app/stores/pageFiltersStore.tsx:104]: internally it seems like there's a mix of calling `this.state` and `this.getState()`. but nbd.

**Review by ryan953 — APPROVED** [review]

File: static/app/stores/pageFiltersStore.tsx (hunk 2 above)
Comment (scttcper) [reply static/app/stores/pageFiltersStore.tsx:104]: ideally, but i think those only exist in classes

Comment (scttcper) [reply static/app/stores/pageFiltersStore.tsx:104]: it is readonly now at least https://github.com/getsentry/sentry/blob/master/static/app/stores/types.tsx#L25

File: static/app/stores/pageFiltersStore.spec.tsx (hunk 1 above)
Comment (scttcper) [reply static/app/stores/pageFiltersStore.spec.tsx:42]: yeah i'm trying to be explicit on what is being tested. fear people would see it fail and change it to toEqual

**Pull request merged**